updater/lib/github_api/dependency_submission.rb

```diff
@@ -134,6 +133,27 @@ def build_manifests(snapshot)
       end
     end
 
+    # For each distinct directory in our manifest list, we only want the highest priority manifests available,
+    # this method will filter out manifests for directories that have lockfiles and so on.
+    sig { params(snapshot: Dependabot::DependencySnapshot).returns(T::Array[Dependabot::DependencyFile]) }
+    def relevant_manifests(snapshot)
+      manifests_by_directory = snapshot.dependency_files.each_with_object({}) do |file, dirs|
```

Comment:
not to be overly clever but what about:

```ruby
snapshot.dependency_files.group_by(&:directory).map { |(dir, files)| files.max_by(&:priority) }
```

?

Reply:
Will address these comments [in my downstream PR](https://github.com/dependabot/dependabot-core/pull/12818)